### PR TITLE
Added  Protobuf extension 

### PR DESCRIPTION
**Pull request by sarabala1979**

### CODE REVIEW

**Comment by xianlubird** [comment]
Does this affect our compatibility?

**Comment by sarabala1979** [comment]
> Does this affect our compatibility?
No
This feature is for Argo API server. This PR will get merge after 2.4 release.

File: workflow/validate/validate.go [hunk 1]
```diff
@@ -579,7 +579,8 @@ func (ctx *templateValidationCtx) validateSteps(scope map[string]interface{}, tm
 			}
 			resolvedTemplates[step.Name] = resolvedTmpl
 		}
-		for i, step := range stepGroup {
+
+		for _, step := range stepGroup.Steps {
```
Comment (jessesuen) [inline workflow/validate/validate.go]: good fix.

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 2]
```diff
@@ -128,221 +128,241 @@ type WorkflowSpec struct {
...
 	// Set scheduler name for all pods.
 	// Will be overridden if container/script template's scheduler name is set.
 	// Default scheduler will be used if neither specified.
 	// +optional
-	SchedulerName string `json:"schedulerName,omitempty"`
+	SchedulerName string `json:"schedulerName,omitempty" protobuf:"bytes,21,opt,name=schedulerName"`
 
 	// PodGC describes the strategy to use when to deleting completed pods
-	PodGC *PodGC `json:"podGC,omitempty"`
+	PodGC *PodGC `json:"podGC,omitempty" protobuf:"bytes,26,opt,name=podGC"`
```
Comment (jessesuen) [inline pkg/apis/workflow/v1alpha1/workflow_types.go]: Why did we jump to 26 here? Can we start out with same ordering?

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 3]
```diff
@@ -128,221 +128,241 @@ type WorkflowSpec struct {
...
 	// SecurityContext holds pod-level security attributes and common container settings.
 	// Optional: Defaults to empty.  See type description for default values of each field.
-	// +optional
-	SecurityContext *apiv1.PodSecurityContext `json:"securityContext,omitempty"`
-
+	// +optiona
+	SecurityContext *apiv1.PodSecurityContext `json:"securityContext,omitempty" protobuf:"bytes,25,opt,name=securityContext"`
 	// PodSpecPatch holds strategic merge patch to apply against the pod spec. Allows parameterization of
 	// container fields which are not strings (e.g. resource limits).
 	PodSpecPatch string `json:"podSpecPatch,omitempty"`
```
Comment (jessesuen) [inline pkg/apis/workflow/v1alpha1/workflow_types.go]: We will need 27 here.

**Pull request merged**